src/main.js

```diff
@@ -319,74 +315,37 @@ var manifestor = function(options) {
   }
 
   function translateTilesources(d, i) {
-    var canvasId = d.canvas.id,
-        mainImageObj = canvasImageStates()[canvasId].mainImageObj;
-
-    var currentBounds = mainImageObj ? mainImageObj.getBounds(true) : null;
+    var canvas = _canvasObjects[d.canvas.id];
 
-    if (currentBounds === null) { return function() { /*no-op*/ }; }
+    if(canvas.hasImageObject) {
+      var currentBounds = canvas.getBounds();
 
-    var xi = d3.interpolate(currentBounds.x, d.canvas.x);
-    var yi = d3.interpolate(currentBounds.y, d.canvas.y);
+      var xi = d3.interpolate(currentBounds.x, d.canvas.x);
+      var yi = d3.interpolate(currentBounds.y, d.canvas.y);
 
-    return function(t) {
-        mainImageObj.setPosition(new OpenSeadragon.Point(xi(t), yi(t)), true);
-        mainImageObj.setWidth(d.canvas.width, true);
-        mainImageObj.setHeight(d.canvas.height, true);
-    };
+      return function(t) {
+        canvas.setPosition(xi(t), yi(t));
+        canvas.setWidth(d.canvas.width);
+      };
+    } else {
+      return function() { /* no-op */ };
+    }
   }
 
   function updateImages(d) {
     var canvasData = d.canvas,
-        canvasImageState = canvasImageStates()[canvasData.id];
+        canvasImageState = _canvasObjects[canvasData.id];
   }
 
   function enterImages(d) {
-
     var canvasData = d.canvas,
-        canvasImageState = canvasImageStates()[canvasData.id];
-
-      viewer.addTiledImage({
-        x: canvasData.x,
-        y: canvasData.y,
-        width: canvasData.width,
-        tileSource: canvasImageState.tileSourceUrl,
-        index: 0, // Add the new image below the stand-in.
-        success: function(event) {
-          addMainImageObj(canvasData.id, event.item);
-          var main = event.item;
-          var tileDrawnHandler = function(event) {
-              viewer.removeHandler('tile-drawn', tileDrawnHandler);
-              main.setOpacity(0,true);
-              fade(main, 1);
-          };
-
-          viewer.addHandler('tile-drawn', tileDrawnHandler);
-        }
-      });
-  }
-
-  function fade(image, targetOpacity, callback) {
-    var currentOpacity = image.getOpacity();
-    var step = (targetOpacity - currentOpacity) / 30;
-    if (step === 0) {
-      callback();
-      return;
-    }
-
-    var frame = function() {
-      currentOpacity += step;
-      if ((step > 0 && currentOpacity >= targetOpacity) || (step < 0 && currentOpacity <= targetOpacity)) {
-        image.setOpacity(targetOpacity);
-        if (callback) callback();
-        return;
-      }
+        canvasImageState = _canvasObjects[canvasData.id];
 
-      image.setOpacity(currentOpacity);
-      OpenSeadragon.requestAnimationFrame(frame);
-    };
-    OpenSeadragon.requestAnimationFrame(frame);
-  };
+    canvasImageState.setPosition(canvasData.x, canvasData.y);
+    canvasImageState.setHeight(canvasData.height);
```

Comment:
One thing to keep in mind is that OSD TiledImages don't support stretching their aspect ratio, so internally they just have a single scale value. This means when you set the height it changes the scale and then when you set the width it changes the scale again. So, at the OSD level it's more efficient to just set one or the other. 

On the other hand, the CanvasObject probably could use both width and height. So maybe having a `setSize` method, that keeps track of both width and height but internally just tells the TiledImage to set its width, would be good.